dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/message/codec/JsonCodec.java

```diff
@@ -40,6 +40,7 @@ public void encode(OutputStream os, Object data, Charset charset) throws EncodeE
         }
     }
 
+    @Override
```

Comment:
**Issue**: The @Override annotation is added to the encode method which enhances readability and ensures proper overriding. However, the method lacks input validation which might lead to EncodeException if null data is passed. <br> **Fix**: Add null checks for the data parameter to avoid potential NullPointerExceptions or EncodeExceptions during runtime. <br> **Code Suggestion**: 
 ```
 if (data == null) {
     throw new IllegalArgumentException("Data cannot be null.");
 }
 // Continue with encoding process
 ```